libs/platform/unit/platform_unit_test.cpp

```diff
@@ -551,31 +551,36 @@ TEST_CASE("trampoline_test", "[platform]")
     const void* helper_functions2[] = {(void*)function_pointer2};
     ebpf_helper_function_addresses_t helper_function_addresses2 = {
         EBPF_COUNT_OF(helper_functions1), (uint64_t*)helper_functions2};
+    ebpf_result_t status = EBPF_SUCCESS;
 
     REQUIRE(ebpf_allocate_trampoline_table(1, &table) == EBPF_SUCCESS);
-    REQUIRE(
-        ebpf_update_trampoline_table(
-            table,
-            EBPF_COUNT_OF(provider_helper_function_ids),
-            provider_helper_function_ids,
-            &helper_function_addresses1) == EBPF_SUCCESS);
-    REQUIRE(
-        ebpf_get_trampoline_function(
-            table, EBPF_MAX_GENERAL_HELPER_FUNCTION + 1, reinterpret_cast<void**>(&test_function)) == EBPF_SUCCESS);
 
-    // Verify that the trampoline function invokes the provider function
+    status = ebpf_update_trampoline_table(
+        table, EBPF_COUNT_OF(provider_helper_function_ids), provider_helper_function_ids, &helper_function_addresses1);
+    if (status != EBPF_SUCCESS) {
```

Comment:
are there cases in the test where this is supposed to fail, or should it be a REQUIRE(status == EBPF_SUCCESS)?

Reply:
ebpf_update_trampoline_table function can return failure due to fault injection code in ebpf_protect_memory() introduced.

Reply:
I think instead of if-else, we should wrap trampoline table pointer in a unique_ptr to ensure it is cleaned up even if test fails.
There is already `ebpf_trampoline_table_ptr` defined in `execution_context_unit_test.cpp`

Reply:
As stated, ebpf_trampoline_table_ptr is defined in .cpp file (not header file). Is there any suggestion of .h file to move the declaration?

Reply:
Why does fault injection happen in the "extension_test" test?  Seems like this case is for non-fault-injection.

Reply:
I don't think it's reasonable to expect that all test cases should pass with fault injection.  All that does it hide bugs we want to catch.  If `ebpf_update_trampoline_table` starts failing for reasons other than fault injection, we need this test to fail so we can catch it.

Reply:
Right, agree. I have reverted the code with changes.